vignettes/preprocess.Rmd

```diff
@@ -0,0 +1,346 @@
+---
+title: "preprocess"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{preprocess}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Before running machine learning, it's often necessary and prudent to preprocess your input data. 
+We provide a function (`preprocess_data`) to preprocess input data. 
+The defaults we chose are based on best practices used in  [FIDDLE](https://gitlab.eecs.umich.edu/mld3/FIDDLE/-/tree/master/).
+Feel free to check out FIDDLE for more information about data preprocessing!
+
+The `preprocess_data` function takes an input dataset where the rows are the samples and the columns are the outcome variable and features.
+We preprocess the data as follows:
+
+- Remove missing outcome values. 
+- Leave binary features as-is (except that categorical variables are converted to 0 and 1, and binary variables with missing features are split into two rows - see below for more details).
+- Normalize continuous features using `caret::preProcess` based on the method provided.
+- Convert categorical features with more than 2 variables to 0 and 1 in multiple columns (one for each variable - not full rank, so each variable has it's own column).
```

Comment:
What do you mean by full rank?
```suggestion
- Convert categorical features with more than 2 variables to 0 and 1 in multiple columns (one for each variable - not full rank, so each variable has it's own column).
```

Reply:
Good point. I looked into it a little more and decided not to use that wording because it's unclear exactly what it means here (it doesn't seem to be the way it's traditionally used in matrix math). Just pushed the change.